pkg/globalmanager/runtime/worker.go

```diff
@@ -269,17 +290,43 @@ func injectDeploymentParam(deployment *appsv1.Deployment, workerParam *WorkerPar
 	if deployment.Spec.Template.Labels == nil {
 		deployment.Spec.Template.Labels = make(map[string]string)
 	}
+	if deployment.Spec.Selector.MatchLabels == nil {
+		deployment.Spec.Selector.MatchLabels = make(map[string]string)
+	}
 
 	for k, v := range generateLabels(object, workerParam.WorkerType) {
 		deployment.Labels[k] = v
 		deployment.Spec.Template.Labels[k] = v
 		deployment.Spec.Selector.MatchLabels[k] = v
 	}
-	deployment.Spec.Template.Spec.Containers[0].Ports = []v1.ContainerPort{
-		{
-			ContainerPort: port,
-		},
+
+	// Edgemesh part, useful for service mapping
+	deployment.Labels["app"] = object.GetName() + "-" + workerParam.WorkerType + "-" + "svc"
+	deployment.Spec.Template.Labels["app"] = object.GetName() + "-" + workerParam.WorkerType + "-" + "svc"
+	deployment.Spec.Selector.MatchLabels["app"] = object.GetName() + "-" + workerParam.WorkerType + "-" + "svc"
+
+	// Ports mapping
+	for _, cont := range deployment.Spec.Template.Spec.Containers {
+		for _, ports := range cont.Ports {
+			if ports.HostPort != 0 {
+				ports.HostPort = port
```

Comment:
Why is the `HostPort` value replaced here?

Reply:
I think it's old code from when it was necessary to specify the `HostPort` value and override it with the service port. I suppose it's not necessary anymore. Can you confirm?

Reply:
I don't really understand the logic behind the port reassignment. If it is not your intention, feel free to remove it